source/Halibut.Tests/TraceLogFileLogger.cs

```diff
@@ -1,25 +1,82 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
 
 namespace Halibut.Tests
 {
     public class TraceLogFileLogger : IDisposable
     {
-        string tempFilePath = Path.GetTempFileName();
+        readonly AsyncQueue<string> queue = new AsyncQueue<string>();
+        readonly string tempFilePath = Path.GetTempFileName();
         string testHash;
 
+        readonly TaskCompletionSource<bool> noMoreDelaysToWritingLogs;
+        readonly Task noMoreDelaysTask;
+        readonly Task writeDataToDiskTask;
+
+        public TraceLogFileLogger()
+        {
+            noMoreDelaysToWritingLogs = new TaskCompletionSource<bool>();
+            noMoreDelaysTask = noMoreDelaysToWritingLogs.Task;
+            writeDataToDiskTask = WriteDataToFile();
+        }
+
         public void SetTestHash(string testHash)
         {
             this.testHash = testHash;
         }
         
         public void WriteLine(string logMessage)
         {
-            File.AppendAllLines(tempFilePath, new[] { logMessage });
+            queue.Enqueue(logMessage);
```

Comment:
The queue here probably does have some locking under the hood. I suspect the time that lock is held for is so small that it wont impact tests, compared to a lock held for the duration of a file append done on every trace or higher log event raised.